pandas/tests/indexes/common.py

```diff
@@ -635,6 +635,16 @@ def test_delete_base(self):
                 # either depending on numpy version
                 result = idx.delete(len(idx))
 
+    def test_equals(self):
+
+        for name, idx in compat.iteritems(self.indices):
+            self.assertTrue(idx.equals(idx))
+            self.assertTrue(idx.equals(idx.copy()))
+            self.assertTrue(idx.equals(idx.astype(object)))
+            self.assertFalse(idx.equals(list(idx)))
```

Comment:
and that is why we have the issue. This shouldn't work. Indexes have meta-data it has to match, and by-definition a list/ndarray/Series cannot.

we have ``.identical`` which is quite strict, ``.equals`` next in strictness. but we already have ``array_equivalent`` which already covers this case (as much less common)